[CI] ClassificationIT testDeleteExpiredData_RemovesUnusedState failing

**Build Scans:**
- [elasticsearch-intake #12402 / part4](https://gradle-enterprise.elastic.co/s/efklpm47mbk3i)
- [elasticsearch-periodic #4727 / openjdk17_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/fgn2q24oz5txy)

**Reproduction Line:**
```
./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.ml.integration.ClassificationIT.testDeleteExpiredData_RemovesUnusedState" -Dtests.seed=EF8C3B43D283D2D9 -Dtests.locale=jgo-Latn-CM -Dtests.timezone=NZ -Druntime.java=22
```

**Applicable branches:**
8.15

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.ClassificationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testDeleteExpiredData_RemovesUnusedState),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Stats were: {"id":"classification_delete_expired_data","state":"started","progress":[{"phase":"reindexing","progress_percent":100},{"phase":"loading_data","progress_percent":100},{"phase":"feature_selection","progress_percent":100},{"phase":"coarse_parameter_search","progress_percent":85},{"phase":"fine_tuning_parameters","progress_percent":0},{"phase":"final_training","progress_percent":0},{"phase":"writing_results","progress_percent":0},{"phase":"inference","progress_percent":0}],"data_counts":{"training_docs_count":100,"test_docs_count":0,"skipped_docs_count":0},"memory_usage":{"timestamp":1730798435661,"peak_usage_bytes":14754,"status":"ok"},"node":{"id":"4TIYx-a-SluSFPEJw_k07w","name":"javaRestTest-1","ephemeral_id":"JgOG8JbAS2ecILe4O1SrUw","transport_address":"127.0.0.1:34485","attributes":{"ml.allocated_processors":"32","ml.max_jvm_size":"536870912","ml.config_version":"12.0.0","ml.machine_memory":"126604156928","ml.allocated_processors_double":"32.0","transform.config_version":"10.0.0","xpack.installed":"true","testattr":"test"}},"assignment_explanation":""}
Expected: <stopped>
     but: was <started>
```

**Issue Reasons:**
- [8.15] 2 failures in test testDeleteExpiredData_RemovesUnusedState (2.3% fail rate in 87 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -370,9 +370,6 @@ tests:
 - class: org.elasticsearch.xpack.security.operator.OperatorPrivilegesIT
   method: testEveryActionIsEitherOperatorOnlyOrNonOperator
   issue: https://github.com/elastic/elasticsearch/issues/119911
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsKeyword
-  issue: https://github.com/elastic/elasticsearch/issues/120071
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldSourceOnlyRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/120080
@@ -481,9 +478,6 @@ tests:
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testSetEnabled
   issue: https://github.com/elastic/elasticsearch/issues/121183
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsInteger
-  issue: https://github.com/elastic/elasticsearch/issues/121473
 - class: org.elasticsearch.upgrades.VectorSearchIT
   method: testBBQVectorSearch {upgradedNodes=1}
   issue: https://github.com/elastic/elasticsearch/issues/121271
@@ -493,15 +487,6 @@ tests:
 - class: org.elasticsearch.xpack.sql.qa.single_node.JdbcDocCsvSpecIT
   method: test {docs.testFilterToday}
   issue: https://github.com/elastic/elasticsearch/issues/121474
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testSingleNumericFeatureAndMixedTrainingAndNonTrainingRows
-  issue: https://github.com/elastic/elasticsearch/issues/121479
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithDatastreams
-  issue: https://github.com/elastic/elasticsearch/issues/121236
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testDeleteExpiredData_RemovesUnusedState
-  issue: https://github.com/elastic/elasticsearch/issues/116234
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testGetProfiles
   issue: https://github.com/elastic/elasticsearch/issues/121101
@@ -525,26 +510,14 @@ tests:
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=ml/*}
   issue: https://github.com/elastic/elasticsearch/issues/120816
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testDependentVariableIsNested
-  issue: https://github.com/elastic/elasticsearch/issues/121458
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testProfileAPIsWhenIndexNotCreated
   issue: https://github.com/elastic/elasticsearch/issues/121096
 - class: org.elasticsearch.xpack.transform.checkpoint.TransformCCSCanMatchIT
   method: testTransformLifecycle_RangeQueryThatMatchesNoShards
   issue: https://github.com/elastic/elasticsearch/issues/121480
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testDependentVariableIsAliasToKeyword
-  issue: https://github.com/elastic/elasticsearch/issues/121492
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsBoolean
-  issue: https://github.com/elastic/elasticsearch/issues/121680
 - class: org.elasticsearch.xpack.migrate.action.ReindexDatastreamIndexTransportActionIT
   issue: https://github.com/elastic/elasticsearch/issues/121737
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testSetUpgradeMode_ExistingTaskGetsUnassigned
-  issue: https://github.com/elastic/elasticsearch/issues/121928
 - class: org.elasticsearch.xpack.security.authc.service.ServiceAccountSingleNodeTests
   method: testAuthenticateWithServiceFileToken
   issue: https://github.com/elastic/elasticsearch/issues/120988

